File: app/src/main/java/in/encrust/writersden/FontAdapter.java
package in.encrust.writersden;

import android.content.Context;
import android.content.Intent;
import android.graphics.Typeface;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.localbroadcastmanager.content.LocalBroadcastManager;
import androidx.recyclerview.widget.RecyclerView;

public class FontAdapter extends RecyclerView.Adapter<FontAdapter.FontHolder> {
    private Context context;
    private List<Typeface> fontList;
    private List<String> fontNameList;

    FontAdapter(Context context, List<Typeface> fontList, List<String> fontNameList) {
        this.context = context;
        this.fontList = fontList;
        this.fontNameList = fontNameList;
    }

    @NonNull
    @Override
    public FontHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.fonts, viewGroup, false);
        return new FontHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull FontHolder fontHolder, final int i) {
        final Typeface font = fontList.get(i);
        final String name = fontNameList.get(i);

        fontHolder.fontView.setTypeface(font);
        fontHolder.fontView.setText(name);
        fontHolder.fontView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent("intent").putExtra("position", i);
                LocalBroadcastManager.getInstance(context).sendBroadcast(intent);
                if (context instanceof Home) {

                    ((Home) context).updateBroadCast(i);
                }
            }
        });

    }

    @Override
    public int getItemCount() {
        return fontList.size();
    }

    class FontHolder extends RecyclerView.ViewHolder {
        TextView fontView;

        FontHolder(@NonNull View itemView) {
            super(itemView);

            fontView = itemView.findViewById(R.id.fonts_fontview);
        }
    }
}

File: app/src/main/java/in/encrust/writersden/FontAdapter.kt
package `in`.encrust.writersden

import android.content.Context
import android.content.Intent
import android.graphics.Typeface
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.localbroadcastmanager.content.LocalBroadcastManager
import androidx.recyclerview.widget.RecyclerView
import java.util.ArrayList

class FontAdapter internal constructor(private val context: Context, private val fontList: ArrayList<Typeface>?, private val fontNameList: List<String>?) : RecyclerView.Adapter<FontAdapter.FontHolder>() {

    override fun onCreateViewHolder(viewGroup: ViewGroup, viewType: Int): FontHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.fonts, viewGroup, false)
        return FontHolder(view)
    }

    override fun onBindViewHolder(fontHolder: FontHolder, i: Int) {
        val font = fontList?.get(i)
        val name = fontNameList?.get(i)

        fontHolder.fontView.typeface = font
        fontHolder.fontView.text = name
        fontHolder.fontView.setOnClickListener {
            val intent = Intent("intent").putExtra("position", i)
            LocalBroadcastManager.getInstance(context).sendBroadcast(intent)
            if (context is Home) {

                context.updateBroadCast(i)
            }
        }

    }

    override fun getItemCount(): Int {
        return fontList!!.size
    }

    inner class FontHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var fontView: TextView

        init {

            fontView = itemView.findViewById(R.id.fonts_fontview)
        }
    }
}
